pkg/cluster/cluster_config.go

```diff
@@ -287,3 +288,32 @@ func IsDefaultAuthMethod(ctx context.Context, cli client.Client) (bool, error) {
 	// we only create userGroups for "IntegratedOAuth" or "" and leave other or new supported type value in the future
 	return authenticationobj.Spec.Type == configv1.AuthenticationTypeIntegratedOAuth || authenticationobj.Spec.Type == "", nil
 }
+
+func GetNodeArchitectures(ctx context.Context, client client.Client) (map[string]struct{}, error) {
```

Comment:
nit: I would probably split this method in two, something like:
- `GetNodes` -> which would return a list of ready nodes
- `GetArchitectures` -> which would use `GetNodes`

Reply:
Good idea, I split the method in the latest commit.